internal/gps/maybe_source.go

```diff
@@ -31,18 +31,21 @@ type maybeSource interface {
 type maybeSources []maybeSource
 
 func (mbs maybeSources) try(ctx context.Context, cachedir string, c singleSourceCache, superv *supervisor) (source, sourceState, error) {
-	var e sourceFailures
+	var errs []error
 	for _, mb := range mbs {
 		src, state, err := mb.try(ctx, cachedir, c, superv)
 		if err == nil {
 			return src, state, nil
 		}
-		e = append(e, sourceSetupFailure{
-			ident: mb.getURL(),
-			err:   err,
-		})
+		errs = append(errs, errors.Wrapf(err, "failed to set up %q", mb.getURL()))
```

Comment:
`errors.WithMessage` is cheaper in cases like this too, when we don't need the stack trace.

Reply:
I'm not going to make that optimization; it doesn't matter and will be net-negative the very first time someone looks to find where this error originates.

Reply:
Doesn't the same logic apply to removing `sourceFailures`?

Reply:
How? `sourceFailures` provides nothing at all.

Reply:
Before this change the final cause in the chain of errors was a `sourceFailures` _with all of it's errors preserved_, so fiddling with the error at the top level, or with the format verb, could potentially provide some insight.

Now it's an `errors.New()` with only the string messages of the original errors, so they'll have to come here to edit this method directly to follow the trail any deeper.

An improved version of `sourceFailures` might implement `Format(s fmt.State, verb rune)`, but I think it's still offering something as-is.